kustomize/internal/commands/edit/add/flagsandargs.go

```diff
@@ -42,6 +44,9 @@ func (a *flagsAndArgs) Validate(args []string) error {
 	if len(a.EnvFileSource) > 0 && (len(a.FileSources) > 0 || len(a.LiteralSources) > 0) {
 		return fmt.Errorf("from-env-file cannot be combined with from-file or from-literal")
 	}
+	if a.Behavior != "" && a.Behavior != "create" && a.Behavior != "replace" && a.Behavior != "merge" {
```

Comment:
It would be preferable if these legal values didn't leak outside of 
https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/generationbehavior.go#L35

Can you add a validation function in that file so out here we either return an error or have a valid instance of the type GenererationBehavior?  There's already a func called NewGenerationBehavior(string), but it doesn't have an error.
You could do a new function like that which returns an error.

Reply:
Can you check the latest commit to see if it has what is needed. If not I can add a new function in types to validate the string.